src/lib/utils/adminSettings/nollning.ts

```diff
@@ -0,0 +1,80 @@
+import { type PrismaClient } from "@prisma/client";
+
+export const NOLLNING_START_KEY = "nollning_start";
+export const NOLLNING_END_KEY = "nollning_end";
+let cache: {
+  value: boolean;
+  lastFetched: Date;
+} | null = null;
+const CACHE_TIME = 3600 * 1000; // 1 hour
+export const isNollningPeriod = async (prisma: PrismaClient) => {
+  const now = new Date();
+  if (
+    cache !== null &&
+    cache.lastFetched.valueOf() + CACHE_TIME > now.valueOf()
+  )
+    return cache.value;
```

Comment:
I do not think we should hard-code the dates. Yes, for the past 3 years we have modified the website code to deal with the nollning, but this might not be the case (as it was not the case the past 15 years before that, the nollning page was it's own website hosted on its own). We don't know what will be done in the future and what DWWW will look like. If we have no skilled DWWWs who know about this, will the stab not be hidden? Seems easier to have an accessible setting that can be switched on/off. Also quicker than waiting for a build when you want to switch the dates.

I also do not think we should remove caching. I can agree that perhaps the caching part should be handled in #403 and not in this PR, but this query would be done on EVERY request to the server, and it is a global value for everyone, it seems dumb to not cache it.

I do, however, to 100% agree it should be authorizedClient calling and not the user's own prisma client, for global caching it should never be done for a user's client. Perhaps we should even create a new client called like "sharedDataClient" which have the privileges of an anonymous user, as to not leak sensitive data. Again, should be handled in #403.

I think we should change it to authorizedClient as of now and look at this again in #403

Reply:
Good points! I'm fine with keeping the caching as is. Using a `authorizedClient` makes the access control a lot more explicit.